bigchaindb/block.py

```diff
@@ -229,3 +232,89 @@ def _start(self):
         p_write.start()
         p_delete.start()
 
+
+class BacklogDeleteRevert(Block):
+
+    def __init__(self, q_backlog_delete):
+        # invalid transactions can stay deleted
+        self.q_tx_to_validate = q_backlog_delete
+        self.q_tx_validated = mp.Queue()
+        self.q_transaction_to_revert = mp.Queue()
+        self.q_tx_delete = mp.Queue()
+
+        self.monitor = Monitor()
+
+    def locate_transactions(self):
+        """
+        Determine if a deleted transaction has made it into a block
+        """
+        # create bigchain instance
+        b = Bigchain()
+
+        while True:
+            tx = self.q_tx_validated.get()
+
+            # poison pill
+            if tx == 'stop':
+                self.q_tx_delete.put('stop')
+                self.q_transaction_to_revert.put('stop')
+                return
+
+            # check if tx is in a (valid) block
+            validity = b.get_blocks_status_containing_tx(tx['id'])
+
+            if validity and list(validity.values()).count(Bigchain.BLOCK_VALID) == 1:
```

Comment:
This will not work like this. There is some period of time since a block is created (and transactions are deleted) and the block is voted valid.

Example timeline:
1. Block1 is createad with tx1 / tx1 is  deleted
2. Receive the delete change for tx1
3. Block is not valid yet / re-insert tx1
4. Block1 is voted valid
5. Block2 is created with tx1